src/jit/morph.cpp

```diff
@@ -6441,17 +6441,17 @@ GenTree* Compiler::fgMorphField(GenTree* tree, MorphAddrContext* mac)
             // Build this tree:  IND(*) #
             //                    |
             //                   ADD(I_IMPL)
-            //                   / \
+            //                   / '\'
             //                  /  CNS(fldOffset)
             //                 /
             //                /
             //               /
             //             IND(I_IMPL) == [Base of this DLL's TLS]
             //              |
             //             ADD(I_IMPL)
-            //             / \
+            //             / '\'
             //            /   CNS(IdValue*4) or MUL
-            //           /                      / \
+            //           /                      / '\'
```

Comment:
This warning is specifically about C language. In C++ context, this sound like a false positive warnig, as language spec says that for a single line comment, compiler shall ignore everything between `//` and EOL characters.

btw, `Ubuntu x64 formatting` leg is failing: https://ci.dot.net/job/dotnet_coreclr/job/master/job/x64_ubuntu_formatting_prtest/28832/consoleFull#1223400814e0cf1bf6-b9e4-4679-b910-ada9d1ab2bcb. This also make these tree diagrams slightly unreadable. How about explicitly supress these warnings with `-Wno-comment` at https://github.com/dotnet/coreclr/blob/721b097db1ac830dafd2590286176ce0e19e84ed/src/pal/tools/gcc-compiler-override.txt#L9

Reply:
Limited the change to src/jit only

Reply:
Why limit it to src/jit only? The same argument applies to all other C++ file.

I did not know that this warning is not relevant to C++. It would be better to disable this globally and undo the similar changes done elsewhere, e.g. https://github.com/dotnet/coreclr/pull/22872/files#diff-86a5e65bfda42d2f380dbb965aab583cR90

Reply:
I'm not sure about ignoring \

https://gcc.gnu.org/ml/gcc/2000-09/msg00381.html

I find mixed messages on the web.

Reply:
> I find mixed messages on the web.

1. Spec
    * https://en.cppreference.com/w/cpp/comment
    * ![image](https://user-images.githubusercontent.com/3840695/54084636-d78d3d80-433b-11e9-95b0-a367415aca24.png)
  ![image](https://user-images.githubusercontent.com/3840695/54084642-eaa00d80-433b-11e9-9666-09b5aa7a07ef.png)

2. Behavior of other compilers
  https://godbolt.org/z/zci7u3 - among the latest versions of `clang`, `gcc`, `msvc` and `icc`, only `gcc` complains about trailing `\` (with `-Wall`); and that is probably because they have inherited it from their C implementation.

Reply:
This doesn't compile with gcc. How about clang?

`
// testing trailing backslash \
int main()
{
  std::cout << "Hello, world!\n";
}
`

Reply:
Clang doesn't compile this snippet either, so it considers the comments multi-line. Btw, you don't need to add apostrophes there. It is enough to put any non-whitespace character at the end of the comment line so that the line doesn't end with the backslash. It seems it would look much nicer that way (we could put in some almost invisible character like a dot or an underscore).